src/tools/cephfs/cephfs-shell

```diff
@@ -1551,15 +1556,44 @@ if __name__ == '__main__':
             args.commands = ['run_script ' +  args.batch, ',quit']
     if args.test:
         args.commands.extend(['-t,'] + [arg + ',' for arg in args.test])
+    if not args.batch and len(args.commands) > 0:
+        args.exe_and_quit = True
+
+    manage_sys_argv(args)
+
+    return args
 
+def manage_sys_argv(args):
+    exe = sys.argv[0]
     sys.argv.clear()
     sys.argv.append(exe)
     sys.argv.extend([i.strip() for i in ' '.join(args.commands).split(',')])
 
+def execute(args):
+    # launch shell else just execute command and exit shell session with the
+    # command's return value.
+    if not args.exe_and_quit:
+        retval = shell.cmdloop()
+        return retval
+
+    args_to_shellcmd = args.commands[0]
+    if len(args.commands) > 1:
+        for cmdarg in args.commands[1:]:
+            args_to_shellcmd += ' ' + cmdarg
+
+    retval = shell.onecmd(args_to_shellcmd)
+    return retval if retval else shell.exit_code
+
+if __name__ == '__main__':
+    config_file = ''
+
+    args = manage_args()
+
     setup_cephfs(config_file)
     shell = CephFSShell()
     shell.exit_code = 0
     read_ceph_conf(shell, config_file)
-    shell.exit_code = shell.cmdloop()
 
-    sys.exit(shell.exit_code)
+    retval = execute(args)
+
+    sys.exit(retval if retval else shell.exit_code)
```

Comment:
```suggestion
    sys.exit(execute(args))
```

Reply:
I separate those two lines purposefully so that while debugging someone can check return value without exiting the debug session.

Reply:
I'll add a comment there for the same.